### PR TITLE
Particle/mixed formulation

### PR DESCRIPTION
**Pull request by lauramoremar**
**📝 Description**
Small modifications for the UP element which allows solving dynamic hypereleastic problems.

**🆕 Changelog**
- Useless "ifs" and variables deleted in the file: updated_lagrangian_UP.cpp
- Size of the mass matrix adapted to the mixed element.
- New test for dynamic cantilever beam added.

### CODE REVIEW

File: applications/ParticleMechanicsApplication/custom_python/particle_mechanics_python_application.cpp [hunk 1]
```diff
@@ -79,6 +79,7 @@ namespace Python{
         KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, IS_PQMPM);
         KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, IS_MAKE_NORMAL_MP_IF_PQMPM_FAILS);
         KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, PQMPM_SUBPOINT_MIN_VOLUME_FRACTION);
+        KRATOS_REGISTER_IN_PYTHON_VARIABLE(m, IS_MIXEDFORMULATION);
```
Comment (VeronikaSinger) [inline applications/ParticleMechanicsApplication/custom_python/particle_mechanics_python_application.cpp]: perhaps we can rename it to IS_MIXED_FORMULATION

File: applications/ParticleMechanicsApplication/python_scripts/mpm_implicit_dynamic_solver.py [hunk 2]
```diff
@@ -40,7 +40,9 @@ def _CreateSolutionScheme(self):
         grid_model_part = self.GetGridModelPart()
         domain_size = self._GetDomainSize()
         block_size  = domain_size
-        if (self.settings["pressure_dofs"].GetBool()):
+        is_mixedformulation = self.settings["pressure_dofs"].GetBool()
```
Comment (VeronikaSinger) [inline applications/ParticleMechanicsApplication/python_scripts/mpm_implicit_dynamic_solver.py]: and here as well is_mixed_formulation

File: applications/ParticleMechanicsApplication/tests/beam_tests/dynamic_UP_hyperelastic_cantilever_test/dynamic_UP_hyperelastic_test_materials.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                "KINEMATIC_HARDENING_MODULUS" : 129240000.0,
+                "REFERENCE_HARDENING_MODULUS" : 450000000.0,
+                "INFINITY_HARDENING_MODULUS"  : 715000000.0,
+                "HARDENING_EXPONENT"          : 16.93,
+                "THICKNESS"                   : 1.0
+            },
+            "Tables"    : {}
+        }
+    }]
+}
```
Comment (VeronikaSinger) [inline applications/ParticleMechanicsApplication/tests/beam_tests/dynamic_UP_hyperelastic_cantilever_test/dynamic_UP_hyperelastic_test_materials.json:19]: if the "Background_Grid.Parts_Grid_Grid_Auto3" is not needed I would delete this part in the materials.json

**Comment by lauramoremar** [comment]
Modifications done! :) Thanks for the review @VeronikaSinger

**Review by VeronikaSinger — APPROVED** [review]
Thanks @lauramoremar

**Pull request merged**